### PR TITLE
TDQ-11688 Propagate changes from the pattern to jobs

### PR DESCRIPTION
**Pull request by yyin-talend**

### CODE REVIEW

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/process/ProcessUpdateManager.java [hunk 1]
```diff
@@ -2164,7 +2165,18 @@ public void run() {
...
+                    } else if(item!=null && "pattern".equalsIgnoreCase(item.getFileExtension())){
+                        //Added TDQ-11688, check pattern update
+                        ITDQPatternService service = null;
+                        try {
+                            service = (ITDQPatternService) GlobalServiceRegister.getDefault().getService(ITDQPatternService.class);
+                        } catch (RuntimeException e) {
+                            // nothing to do
+                        }
+                        if (service != null) {
+                            service.updateJobForPattern(node,item);
```
Comment (ggu-talend) [inline main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/process/ProcessUpdateManager.java]: Shouldn't do update parameters or such here, should be do it in one command, for example. UpdateNodeParameterCommand

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/properties/controllers/ControllerRepositoryValueHander.java [hunk 2]
```diff
@@ -131,6 +134,18 @@ private void fastRepositoryUpdateProperty(IElementParameter param) {
                             + lastLinkItem.getProperty().getLabel();
 
                 }
+                if(StringUtils.isEmpty(displayName)){
+                    ITDQPatternService tdqService = null;
+                    try {
+                        tdqService = (ITDQPatternService) GlobalServiceRegister.getDefault().getService(ITDQPatternService.class);
```
Comment (ggu-talend) [inline main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/properties/controllers/ControllerRepositoryValueHander.java]: before get the service, I prefer using "if(GlobalServiceRegister.getDefault().isServiceRegistered(ITDQPatternService.class)) {...}" to check first.

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/properties/controllers/PropertyTypeController.java [hunk 3]
```diff
@@ -306,7 +311,9 @@ protected Command createButtonCommand(Button button) {
                     String[] listRepositoryItems = dbTypeParam.getListRepositoryItems();
                     dialog = new RepositoryReviewDialog(Display.getCurrent().getActiveShell(), ERepositoryObjectType.METADATA,
                             param.getRepositoryValue(), listRepositoryItems);
-                } else {
+                }else if(elem.getElementName().startsWith("tPattern")){
```
Comment (ggu-talend) [inline main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/properties/controllers/PropertyTypeController.java]: if it's for component, I prefer something like "if (elem instanceof INode &&((INode)elem).getComponent().getName().startsWith("tPattern")) {.."

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/properties/controllers/ControllerRepositoryValueHander.java (hunk 2 above)
Comment (yyin-talend) [inline main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/properties/controllers/ControllerRepositoryValueHander.java]: changed

**Comment by yyin-talend** [comment]
Modified

**Pull request merged**